src/Hazelcast.Net/DistributedObjects/Impl/HVectorCollection.cs

```diff
@@ -67,22 +68,39 @@ public async Task<IVectorDocument<TVal>> PutIfAbsentAsync(TKey key, IVectorDocum
             var rawResponse = VectorCollectionPutIfAbsentCodec.DecodeResponse(response).Value;
             return await DeserializeIVectorDocumentAsync<TVal>(rawResponse).CfAwait();
         }
-        public Task PutAllAsync([NotNull] IDictionary<TKey, IVectorDocument<TVal>> vectorDocumentMap)
+        public async Task PutAllAsync([NotNull] IDictionary<TKey, IVectorDocument<TVal>> vectorDocumentMap)
         {
             vectorDocumentMap.ThrowIfNull();
+            var entriesByPartition = new Dictionary<int, List<KeyValuePair<IData, IVectorDocument<IData>>>>();
 
-            var rawEntries = new List<KeyValuePair<IData, IVectorDocument<IData>>>();
             foreach (var kvp in vectorDocumentMap)
             {
                 var key = kvp.Key ?? throw new ArgumentException($"Key cannot be null in {nameof(vectorDocumentMap)}.");
                 var val = kvp.Value ?? throw new ArgumentException($"Value cannot be null in {nameof(vectorDocumentMap)}.");
-                var dataKey = ToSafeData(key);
-                var dataValue = ToSafeData(val);
-                var rawDocument = new VectorDocument<IData>(dataValue, kvp.Value.Vectors);
-                rawEntries.Add(new KeyValuePair<IData, IVectorDocument<IData>>(dataKey, rawDocument));
+                var(dataKey, rawDocument) = PrepareForPut(key, val);
+                var partitionId = Cluster.Partitioner.GetPartitionId(dataKey.PartitionHash);
+
+                if (entriesByPartition.TryGetValue(partitionId, out var list))
+                {
+                    list.Add(new KeyValuePair<IData, IVectorDocument<IData>>(dataKey, rawDocument));
+                }
+                else
+                {
+                    entriesByPartition[partitionId] = new List<KeyValuePair<IData, IVectorDocument<IData>>>()
+                    {
+                        new KeyValuePair<IData, IVectorDocument<IData>>(dataKey, rawDocument)
+                    };
```

Comment:
(asking just to learn)
What does this code do?

Reply:
In some languages there is a shortcut to initialize all items of an array with some values.
Does this code do something like that?
What's the length of the list when you don't specify its size?